go-fuzz/main.go

```diff
@@ -99,7 +99,11 @@ func main() {
 			// Try the default. Best effort only.
 			var bin string
 			cfg := new(packages.Config)
-			cfg.Env = append(os.Environ(), "GO111MODULE=off")
+			// Preliminary modules support: note that we do not set 
+			// GO111MODULE here in order to respect any GO111MODULE setting by the user
+			// as we are finding dependencies. See preliminary modules support 
+			// comments in go-fuzz-build/main.go for more details.
+			cfg.Env = os.Environ()
```

Comment:
like before, I think you can just leave this as the default.

Reply:
I hear what you are saying, and for most projects I would agree with you that this would be unnecessary.

That said, for this project in particular, and for this problem in particular, I think it might be more prudent to leave it for whoever needs to investigate and experiment to resolve some future modules-related breakage. (That future person might be myself, or might be one of the people commenting on this PR / recent go-fuzz modules discussions, or maybe a random gopher who hit a future problem with go-fuzz, or it might be Dmitry, especially if far enough into the future).

Solving future modules-related breakage will likely require some non-trivial understanding across multiple domains, including; 1. details of how modules work, which is still a moving target, 2. internals of go-fuzz, 3. probably some knowledge of go/packages, and 4. the details, environment and setup for whatever symptoms are being reported on some future go-fuzz issue.

There are a very small number of people who would simultaneously score well on 1-3 (and at least to start, it might take some iterations for someone other than the reporter to understand number 4. for a particular problem report).

I think leaving this here helps someone investigating some related failure in the future, including if it someone that does not have deep enough knowledge of all the moving pieces (or, if it ends up being me after a sufficiently long enough time has passed ;-)

Reply:
You could have just said "this is all very brittle so I want to be extremely explicit about the environment" :) SGTM